tests/Test/src/basic/test_basic_logsumexp.birch

```diff
@@ -0,0 +1,149 @@
+
+/*
+ * Test log-sum-exp implementations in `log_sum_exp` and
+ * `resample_reduce`.
+ */
+program test_logsumexp() {
+  // Generate random weights
+  w:Real[100];
+  for n in 1..100 {
+    w[n] <- simulate_gaussian(0.0, 1.0);
+  }
+
+  // Compare with common two-pass algorithm
+  let y <- log_sum_exp_twopass(w);
+  let w2 <- transform(w, \(x:Real) -> { return 2*w; });
+  let ess <- y*y/log_sum_exp_twopass(w2);
+  if !check_ess_logsumexp(w, ess, y) {
+    exit(1);
+  }
+  
+  // Check overflow
+  let v <- transform(w, \(x:Real) -> { return x + 1000.0; });
+  let v2 <- transform(v, \(x:Real) -> { return 2*w;} );
+  y <- y + 1000.0;
+  ess <- y*y/log_sum_exp_twopass(v2);
+  if !check_ess_logsumexp(w, ess, y) {
+    exit(1);
+  }
+
+  // Check underflow
+  w <- [1e-20, log(1e-20)];
+  y <- 1e-20;
+  ess <- 1.0;
+  if !check_ess_logsumexp(w, ess, y) {
+    exit(1);
+  }
+
+  // Check empty input
+  x:Real[0];
+  if !check_ess_logsumexp(x, 0.0, -inf) {
+    exit(1);
+  }
+
+  // Special cases involving -inf, inf, and nan.
+  let cases <- [
+    ([-inf, -inf], nan, -inf),
+    ([-inf, nan], nan, -inf),
+    ([nan, -inf], nan, -inf),
+    ([-inf, 42.0], 1.0, 42.0),
+    ([nan, 42.0], 1.0, 42.0),
+    ([42.0, -inf], 1.0, 42.0),
+    ([42.0, nan], 1.0, 42.0),
+    ([-inf, inf], 1.0, inf),
+    ([nan, inf], 1.0, inf),
+    ([42.0, inf], 1.0, inf),
+    ([inf, -inf], 1.0, inf),
+    ([inf, nan], 1.0, inf),
+    ([inf, 42.0], 1.0, inf),
+    ([inf, inf], 1.0, inf),
+  ];
+  for n in 1..length(cases) {
+    (w, y, ess) <- cases[n];
+    if !check_ess_logsumexp(w, ess, y) {
+      exit(1);
+    }
+  }
+}
+
+/*
+ * Exponentiate and sum a log weight vector.
+ *
+ * @param w Log weights.
+ *
+ * @return the logarithm of the sum.
+ *
+ * !!! note
```

Comment:
Make it `@note`, but keep the following paragraph indented (it converts to `!!! note` in Markdown anyway, but standardizing on Doxygen-style `@note` toward C++ convergence).

Reply:
```suggestion
 * @note
```